name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/PostbankPDFExtractor.java

```diff
@@ -1,283 +1,325 @@
-        return true;
-    }
-
-    private void processFeeEntries(Object t, Map<String, String> v, DocumentType type)
-    {
-        if (t instanceof name.abuchen.portfolio.model.Transaction)
-        {
-            Money fee = Money.of(asCurrencyCode(v.get("currency")), asAmount(v.get("fee")));
-            PDFExtractorUtils.checkAndSetFee(fee, (name.abuchen.portfolio.model.Transaction) t, type);
-        }
-        else
-        {
-            Money fee = Money.of(asCurrencyCode(v.get("currency")), asAmount(v.get("fee")));
-            PDFExtractorUtils.checkAndSetFee(fee,
-                            ((name.abuchen.portfolio.model.BuySellEntry) t).getPortfolioTransaction(), type);
-        }
-    }
-}
+package name.abuchen.portfolio.datatransfer.pdf;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Map;
+
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Block;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.DocumentType;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Transaction;
+import name.abuchen.portfolio.model.AccountTransaction;
+import name.abuchen.portfolio.model.BuySellEntry;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.model.Transaction.Unit;
+import name.abuchen.portfolio.money.Money;
+
+@SuppressWarnings("nls")
+public class PostbankPDFExtractor extends AbstractPDFExtractor
+{
+    private static final String FLAG_WITHHOLDING_TAX_FOUND = "exchangeRate"; //$NON-NLS-1$
+
+    public PostbankPDFExtractor(Client client)
+    {
+        super(client);
+
+        addBankIdentifier("Postbank"); //$NON-NLS-1$
+
+        addBuySellTransaction();
+        addDividendeTransaction();
+    }
+
+    @Override
+    public String getLabel()
+    {
+        return "Postbank"; //$NON-NLS-1$
+    }
+
+    private void addBuySellTransaction()
+    {
+        DocumentType type = new DocumentType("Wertpapier Abrechnung (Kauf|Verkauf|Ausgabe Investmentfonds|R.cknahme Investmentfonds)");
+        this.addDocumentTyp(type);
+
+        Transaction<BuySellEntry> pdfTransaction = new Transaction<>();
+        pdfTransaction.subject(() -> {
+            BuySellEntry entry = new BuySellEntry();
+            entry.setType(PortfolioTransaction.Type.BUY);
+            return entry;
+        });
+
+        Block firstRelevantLine = new Block("Wertpapier Abrechnung (Kauf|Verkauf|Ausgabe Investmentfonds|R.cknahme Investmentfonds).*");
+        type.addBlock(firstRelevantLine);
+        firstRelevantLine.set(pdfTransaction);
+
+        pdfTransaction
+
+                        // Is type --> "Verkauf" change from BUY to SELL
+                        .section("type").optional().match("Wertpapier Abrechnung (?<type>(Verkauf|R.cknahme Investmentfonds)).*").assign((t, v) -> {
+                            if (v.get("type").equals("Verkauf") || v.get("type").contains("cknahme Investmentfonds"))

```

Comment:
Maybe change this to this?
`if (v.get("type").equals("Verkauf") || v.get("type").equals("Rücknahme Investmentfonds)"))`

Reply:
I wanted to avoid the 'ü' because of potential encoding problems. That's why in the Regex there's also "R.cknahme".